src/StoreApi/Utilities/OrderController.php

```diff
@@ -431,15 +431,71 @@ protected function validate_coupon_usage_limit( \WC_Coupon $coupon, \WC_Order $o
 		$coupon_usage_limit = $coupon->get_usage_limit_per_user();
 
 		if ( $coupon_usage_limit > 0 ) {
```

Comment:
We should return early here rather than nest the entire function below.

Reply:
So maybe removing the existing check and doing this?

```
if ( $coupon_usage_limit === 0 ) {
	return;
}
```

early on?